pkg/services/ngalert/schedule/schedule.go

```diff
@@ -139,18 +154,164 @@ func (sch *schedule) Unpause() error {
+	sch.log.Info("start of admin configuration sync")
+	orgIds, err := sch.adminConfigStore.GetOrgsWithAdminConfiguration()
+	if err != nil {
+		return err
+	}
+
+	q := &models.GetOrgAdminConfiguration{OrgIDs: orgIds}
+	if err := sch.adminConfigStore.GetAdminConfigurations(q); err != nil {
+		return err
+	}
+
+	sch.log.Info("found admin configurations", "count", len(q.Result))
+
+	orgsFound := make(map[int64]struct{}, len(q.Result))
+	sch.sendersMtx.Lock()
+	for _, cfg := range q.Result {
+		orgsFound[cfg.OrgID] = struct{}{} // keep track of the which senders we need to keep.
+
+		existing, ok := sch.senders[cfg.OrgID]
+
+		// If the tenant has no Alertmanager(s) configured and no running sender no-op.
+		if !ok && len(cfg.Alertmanagers) == 0 {
+			sch.log.Debug("no external alertmanagers configured", "org", cfg.OrgID)
+			continue
+		}
+
+		// We have a running sender but no Alertmanager(s) configured, shut it down.
+		if ok && len(cfg.Alertmanagers) == 0 {
+			sch.log.Debug("no external alertmanager(s) configured, sender will be stopped", "org", cfg.OrgID)
+			delete(orgsFound, cfg.OrgID)
+			continue
+		}
+
+		// We have a running sender, check if we need to apply a new config.
+		if ok {
+			if sch.sendersCfg[cfg.OrgID] == cfg.AsSHA256() {
+				sch.log.Debug("sender configuration is the same as the one running, no-op", "org", cfg.OrgID, "alertmanagers", cfg.Alertmanagers)
+				continue
+			}
+
+			sch.log.Debug("applying new configuration to sender", "org", cfg.OrgID, "alertmanagers", cfg.Alertmanagers)
+			err = existing.ApplyConfig(cfg)
+			if err != nil {
+				sch.log.Error("failed to apply configuration", "err", err, "org", cfg.OrgID)
+			}
+			continue
+		}
+
+		// No sender and have Alertmanager(s) to send to - start a new one.
+		sch.log.Info("creating new sender for the external alertmanagers", "org", cfg.OrgID, "alertmanagers", cfg.Alertmanagers)
+		s, err := sender.New(sch.metrics)
+		if err != nil {
+			sch.log.Error("unable to start the sender", "err", err, "org", cfg.OrgID)
+			continue
+		}
+
+		sch.senders[cfg.OrgID] = s
+		s.Run()
+
+		err = s.ApplyConfig(cfg)
+		if err != nil {
+			sch.log.Error("failed to apply configuration", "err", err, "org", cfg.OrgID)
+			continue
+		}
+
+		sch.sendersCfg[cfg.OrgID] = cfg.AsSHA256()
+	}
+	sch.sendersMtx.Unlock()
+
+	sendersToStop := map[int64]*sender.Sender{}
+
+	sch.sendersMtx.Lock()
+	for orgID, s := range sch.senders {
+		if _, exists := orgsFound[orgID]; !exists {
+			sendersToStop[orgID] = s
+			delete(sch.senders, orgID)
+			delete(sch.sendersCfg, orgID)
+		}
+	}
+	sch.sendersMtx.Unlock()
+
+	// We can now stop these senders w/o having to hold a lock.
+	for orgID, s := range sendersToStop {
+		sch.log.Info("stopping sender", "org", orgID)
+		s.Stop()
+		sch.log.Info("stopped sender", "org", orgID)
+	}
+
+	sch.log.Info("finish of admin configuration sync")
+
+	return nil
+}
+
+// AlertmanagersFor returns all the discovered Alertmanager(s) for a particular organization.
```

Comment:
This method and `DroppedAlertmanagersFor` are not used except in tests for now. I'm going to build an endpoint that communicates these back in a follow-up PR but need them for testing here.